README.md

```diff
@@ -44,70 +45,133 @@ Testing
+You can also run `vagrant ssh` to connect to the VM (or open the VirtualBox console and interface with the desktop
+environment) and run the tests manually if you wish.
 
 Usage
 -----
 
-To use any of the gpii components or functionality in node.js, use the
+To use any of the gpii components or functionality in Node.js, use the
 following statements to get access to fluid and/or gpii objects.
 
-    var fluid = require("universal"),
-        gpii = fluid.registerNamespace("gpii");
-    // Now you will have access to both fluid and gpii namespaces.
+```
+var fluid = require("universal"),
+    gpii = fluid.registerNamespace("gpii");
+// Now you will have access to both fluid and gpii namespaces.
+```
 
 
 #### Running node-based production tests
-The purpose of these tests are to test production config setups of the system. This involves using the online preferences server when fetching preferences sets, so there are extended requirements for these tests.
+The purpose of these tests are to test production config setups of the system. This involves using the online
+preferences server when fetching preferences sets, so there are extended requirements for these tests.
 
-These tests are a supplement to the `all-tests.js` (and hence not part of that test suite) and should be run separately when testing the system and having the below requirements available.
+These tests are a supplement to the `all-tests.js` (and hence not part of that test suite) and should be run separately
+when testing the system and having the below requirements available.
 
 Requirements:
 * an internet connection
 * a preferences server running at `http://preferences.gpii.net` containing at least the following (unmodified) NP set: `MikelVargas`
 
 The tests are run using the following command:
 
-    node tests/ProductionConfigTests.js
+`node tests/ProductionConfigTests.js`
+
+#### Coverage Reporting
+
+The preferred way to consistentely generate a code coverage report is to use Vagrant as described above.  When you 
+start a VM using `vagrant up` and run `npm run test:vagrant`, the full test suite will run in the VM,  and a coverage 
+report will be saved to the `reports` directory.  You can also run the `npm test` command on your local machine, but 
+you will need to ensure that browsers receive focus when they are launched (see above).
 
+The `npm test` command has [two additional associated scripts](https://docs.npmjs.com/misc/scripts).  The `pretest`
+script runs before the command defined for the `test` script.  The `posttest` script runs after.  In our case
+we use a `pretest` script to clean up previous coverage data before we run the tests, and a `posttest` script to 
+compile the actual report.  You should not need to run the pretest scripts manually before running either the node or
```

Comment:
You said "You should not need to run the `pretest` scripts manually". What about `posttest`?